### PR TITLE
D1: Update local-development.md

### PR DESCRIPTION
**Pull request by elithrar**
Add Miniflare testing details to the D1 docs per https://miniflare.dev/storage/d1

### CODE REVIEW

File: content/d1/learning/local-development.md [hunk 1]
```diff
@@ -102,6 +102,45 @@ Users of wrangler `2.x` must use the `--persist` flag: previous versions of wran
 
 ## Test programmatically
 
+### Miniflare
+
+[Miniflare](https://miniflare.dev/) allows you to simulate a Workers and resources like D1 using the same underlying runtime and code as used in production.
+
+You can use Miniflare's [support for D1](https://miniflare.dev/storage/d1) to create D1 databases you can use for testing:
+
+import ConfigTabs from "../components/mdx/config-tabs";
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/local-development.md]:
```suggestion
import ConfigTabs from "../components/mdx/config-tabs";
```

Comment (Maddy-Cloudflare) [reply content/d1/learning/local-development.md]: Is this sentence complete?

File: content/d1/learning/local-development.md [hunk 2]
```diff
@@ -102,6 +102,45 @@ Users of wrangler `2.x` must use the `--persist` flag: previous versions of wran
...
+
+```js
+const mf = new Miniflare({
+  d1Databases: ["DB"],
+});
+```
+
+</ConfigTabs>
+
+You can then use the `getD1Database` method to retrieve the simulated database and run queries against it as if it were your real production D1 database:
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/local-development.md]:
```suggestion
You can then use the `getD1Database()` method to retrieve the simulated database and run queries against it as if it were your real production D1 database:
```

**Review by Maddy-Cloudflare — COMMENTED** [review]
There's a sentence that seems incomplete.

File: content/d1/learning/local-development.md (hunk 1 above)
Comment (rozenmd) [reply content/d1/learning/local-development.md]: ~That should probably be in a code snippet~

Oh wait, that's MDX from [miniflare's docs](https://github.com/cloudflare/miniflare/blob/master/docs/src/content/storage/d1.md) - I don't think that'll work in cloudflare-docs @elithrar

**Review by Maddy-Cloudflare — APPROVED** [review]

**Pull request merged**